### PR TITLE
ES6 Class Syntax refactor

### PR DESCRIPTION
**Pull request by sbernaldev**
Hi it's me again! I saw that this example is not using ES6 syntax at all, so I decided to update it and I'm waiting for your OK!

Regards

### CODE REVIEW

**Comment by i0natan** [comment]
@sbernaldev Welcome back! :] If you're satisfied with this type of challenges (great way to learn), we plan to add ES6 and TS examples to all bullets

@BrunoScheufler @sagirk @js-kyle 
Going by our new workflow, I randomly entered the repo today, saw a new PR, [looked at the table](https://github.com/i0natan/nodebestpractices/blob/master/.operations/operations-manual.md), assigning to Bruno (in charge of language syntax). Makes sense?

File: sections/errorhandling/operationalvsprogrammererror.md [hunk 1]
```diff
@@ -8,16 +8,18 @@ Distinguishing the following two error types will minimize your app downtime and
 
 ```javascript
 // marking an error object as operational 
-var myError = new Error("How can I add new product when no value provided?");
+const myError = new Error("How can I add new product when no value provided?");
 myError.isOperational = true;
 
 // or if you're using some centralized error factory (see other examples at the bullet "Use only the built-in Error object")
-function appError(commonType, description, isOperational) {
+class appError {
```
Comment (BrunoScheufler) [inline sections/errorhandling/operationalvsprogrammererror.md]: When using classes, I'd prefer to capitalize the first letter as a part of common popular code conventions (AirBnB, standard). The name would have to changed down below too 👍 

`appError` => `AppError`

**Comment by sagirk** [comment]
@sbernaldev Welcome back! Thanks for modernizing this example. 💯 

@i0natan Yup, all good! Makes sense. 👍

**Comment by sbernaldev** [comment]
> @sbernaldev Welcome back! :] If you're satisfied with this type of challenges (great way to learn), we plan to add ES6 and TS examples to all bullets

For me it's fine! I'm not familiar with TS but I would be glad as well. One question, the idea is ho have ES5 + ES6 + TS or just ES6 + TS?

**Comment by BrunoScheufler** [comment]
@sbernaldev I think we're striving for the latest and greatest JavaScript features and also TyoeScript, so it would be one example with both ES(latest) + TS

**Comment by BrunoScheufler** [comment]
You do not have to add TypeScript though, as this exceeds the current scope and we might do that later on for all sections at once :+1:

File: sections/errorhandling/operationalvsprogrammererror.md [hunk 2]
```diff
@@ -8,16 +8,18 @@ Distinguishing the following two error types will minimize your app downtime and
...
+  constructor (commonType, description, isOperational) {
     Error.call(this);
     Error.captureStackTrace(this);
     this.commonType = commonType;
     this.description = description;
     this.isOperational = isOperational;
+  }
 };
 
 throw new appError(errorManagement.commonErrors.InvalidInput, "Describe here what happened", true);
```
Comment (BrunoScheufler) [inline sections/errorhandling/operationalvsprogrammererror.md]: This would have to be capitalized as well :+1:

Comment (BrunoScheufler) [reply sections/errorhandling/operationalvsprogrammererror.md]: Oh, the selection seems a bit off, I meant the `throw new appError` line at the bottom

Comment (sbernaldev) [reply sections/errorhandling/operationalvsprogrammererror.md]: Done!

**Comment by sagirk** [comment]
Congratulations @sbernaldev! Landed in https://github.com/i0natan/nodebestpractices/commit/a94dd7e70de79b8d30c0c36ecd85c5c0401d33de. 🎉

**Pull request merged**